build/ci/Dockerfile

```diff
@@ -0,0 +1,41 @@
+FROM ubuntu:22.04 as ci
+
+ARG REF_ARG=automation_pipeline/ghcr
+ARG SSH_KEY_ARG="not a key"
+ARG NR_LICENSE_KEY_ARG="provided by task definition"
+ARG NEW_RELIC_API_KEY_ARG="not a key"
+ARG NEW_RELIC_ACCOUNT_ID_ARG="not a key"
+ARG ANSIBLE_INVENTORY_FOLDER_ARG="/srv/runner/inventory/$REF_ARG"
+ARG ANSIBLE_INVENTORY_FILE_ARG="inventory.ec2"
+
```

Comment:
I think that the variables provided on execution time should be only in ENV and not in ARG 
https://docs.docker.com/engine/reference/builder/#arg
https://docs.docker.com/engine/reference/builder/#env

If are passed in ARG they will be stored with the image and if I'm not mistaken we just want most (if not all) to be on runtime

So i.e. 
```
ENV ANSIBLE_INVENTORY_FOLDER=/srv/runner/inventory/automation_pipeline/ghcr
```

will have that default value if no overrides are passed on docker run and it can be overridden with
```
docker run -e ANSIBLE_INVENTORY_FOLDER=/another/path
```